packages/server/src/sdk/workspace/ai/agents/utils.ts

```diff
@@ -0,0 +1,70 @@
+import { createToolSource } from "packages/server/src/ai/tools/base/ToolSourceRegistry"
+import { Agent, Tool } from "@budibase/types"
+import { ai } from "@budibase/pro"
+
+export interface BuildPromptAndToolsOptions {
+  baseSystemPrompt?: string
+  includeGoal?: boolean
+}
+
+export async function buildPromptAndTools(
+  agent: Agent,
+  options: BuildPromptAndToolsOptions = {}
+): Promise<{
+  systemPrompt: string
+  tools: Tool[]
+}> {
+  const { baseSystemPrompt, includeGoal = true } = options
+
+  let toolGuidelines = ""
+  const allTools: Tool[] = []
+
+  for (const toolSource of agent.allowedTools || []) {
+    const toolSourceInstance = createToolSource(toolSource)
+
+    if (!toolSourceInstance) {
+      continue
+    }
+
+    const guidelines = toolSourceInstance.getGuidelines()
+    if (guidelines) {
+      const prefix = `When using ${toolSourceInstance.getName()} tools, ensure you follow these guidelines:\n${guidelines}`
```

Comment:
NAB: Prompt injection? If this name is user supplied.

Reply:
Only admins configure these, so I don't think it's a concern really.